test/python/algorithms/test_adapt_vqe.py

```diff
@@ -0,0 +1,141 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Test of the AdaptVQE minimum eigensolver """
+import unittest
+from test.python.algorithms import QiskitAlgorithmsTestCase
+from qiskit.algorithms.minimum_eigen_solvers.vqe import VQE
+from qiskit.opflow.gradients.gradient import Gradient
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.circuit.library import EvolvedOperatorAnsatz
+from qiskit.quantum_info import SparsePauliOp
+from qiskit.opflow import PauliSumOp
+from qiskit.utils import algorithm_globals, QuantumInstance
+from qiskit import BasicAer
+from qiskit.algorithms.minimum_eigen_solvers.adapt_vqe import AdaptVQE
+
+
+class TestAdaptVQE(QiskitAlgorithmsTestCase):
```

Comment:
How long do these tests run? I agree it's nice to have a test that checks the whole algorithm end-to-end, but we should also have a suite of tests that run very fast (not more than a few seconds in total) that check each part of the code. Could we add some test that only check individual parts too? Like
* check all errors are properly raised
* check the gradient calculation works on a tiny toy example
* check the different termination criteria are reached 
* check the VQE is still the same at the end 🙂

Reply:
For clarification, by
> check the gradient calculation works on a tiny toy example

I meant checking that we implemented the calculation via commutator correctly. For instance we could check this by having only a single excitation operator, say `X` and check that the calculation of the gradient via `[H, X]` gives the same as we compute by hand:
```
d/d theta <psi(theta)|H|psi(theta)>, |psi(theta)> = RX(2 * theta)|0>
```